src/main/java/org/wildfly/security/evidence/BearerTokenEvidence.java

```diff
@@ -0,0 +1,21 @@
+package org.wildfly.security.evidence;
+
+import org.wildfly.common.Assert;
+
+/**
+ * An {@link Evidence} that usually holds a bearer security token.
+ *
+ * @author <a href="mailto:psilva@redhat.com">Pedro Igor</a>
+ */
+public class BearerTokenEvidence implements Evidence {
```

Comment:
I think this is not sufficient because evidence (like credentials) relies strongly on exact type checking.  If you have a concrete class which is usable, with a usable subclass as well, it could cause some checks to get confused if they check for the supertype before a more specific mechanism checks for the subtype.  For this reason, each type of evidence should be distinct and final.

However, shared functionality can be factored out into base abstract classes or interfaces (see the Algorithm* types and the X509 types for examples of this).